src/main/java/easton/Easton.java

```diff
@@ -0,0 +1,263 @@
+package easton;
+
+import easton.model.Deadline;
+import easton.model.Event;
+import easton.model.Task;
+import easton.model.ToDo;
+import easton.view.Ui;
+
+import java.io.*;
+import java.util.ArrayList;
+
+public class Easton {
+
+    private ArrayList<Task> tasks = new ArrayList<>();;
+    private Storage storage;
+    private Ui<Task> ui;
+
+    public Easton(String fileName) {
+        try {
+            storage = new Storage(fileName);
+        } catch (IOException e) {
+            Ui.displayText("Cannot connect to the storage.");
+        }
+
+        tasks = retrieveTasks();
+        ui = new Ui<>();
+    }
+
+    public void run() {
+        Ui.welcome();
+        boolean isFinished = false;
+        Action action;
+        String userInput;
+
+        while (!isFinished) {
+            userInput = ui.input();
+            Ui.divider();
+
+            try {
+                action = getActionFromInput(userInput);
+            } catch (IllegalActionException e) {
+                Ui.displayText(e.getMessage());
+                action = Action.INVALID;
+            }
+
+            switch (action) {
```

Comment:
Great job with the indentation of the switch statement. Many students often overlook this part so good job for avoiding this mistake!